### PR TITLE
feat: migrate to typescript

### PR DESCRIPTION
**Pull request by palaparthi**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "outDir": "./built",
```
Comment (dkundel) [inline tsconfig.json]: Can we change this one to `./dist`?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "target": "es2018",
+    "strict": true,
+    "strictNullChecks": true,
+    "alwaysStrict": true,
+    "moduleResolution": "node"
+  },
+  "include": [
+    "./*"
```
Comment (dkundel) [inline tsconfig.json]: let's move this to `./src` instead. So `./src/main.ts` and `./dist/main.js`

**Comment by palaparthi** [comment]
Can you take a look and let me know if I am missing something. Thanks!

**Review by KILDIAN — CHANGES_REQUESTED** [review]
>

**Pull request merged**